common/exclusiveStringMap.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package common
+
+import (
+	"github.com/pkg/errors"
+	"strings"
+	"sync"
+)
+
+type ExclusiveStringMap struct {
+	lock          *sync.Mutex
+	m             map[string]struct{}
+	caseSensitive bool
+}
+
+func NewExclusiveStringMap(fromTo FromTo, goos string) *ExclusiveStringMap {
```

Comment:
>goos string [](start = 42, length = 11)

On one hand, this feels like a really odd choice. On the other hand, I understand that it's for testing, and we need to work around the fact that runtime.GOOS is a const not a var.

WRT that, consider doing strings.EqualFold for goos == "windows"/"darwin", as we could oddly capitalize it in tests by accident.

Reply:
Yeah, it felt odd to me too, and I did it for exactly the reason you guessed.  I've followed your suggestion.  Can you take quick look to sanity check my commit please?